proto/deadlock.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+package deadlockpb;
+
+import "gogoproto/gogo.proto";
+
+message WaitForEntriesRequest {
+}
+
+message WaitForEntriesResponse {
+	repeated WaitForEntry entries = 1 [(gogoproto.nullable) = false];
+}
+
+message WaitForEntry {
+	// The transaction id that is waiting.
+	uint64 txn = 1;
+	// The transaction id that is being waited for.
+	uint64 wait_for_txn = 2;
+	// The hash value of the key is being waited for.
+	uint64 key_hash = 3;
+}
+
+enum DeadlockRequestType {
+	Detect = 0;
+	// CleanUpWaitFor cleans a single entry the transaction is waiting.
+	CleanUpWaitFor = 1;
+	// CleanUp cleans all entries the transaction is waiting.
+	CleanUp = 2;
```

Comment:
Maybe CleanUpAll?

Reply:
`CleanUpAll` can be misunderstood as clean up all transactions in the detector.